### ISSUE TITLE
Click + drag not fully solved yet

### ISSUE THREAD

**Issue by ryanbugden** [ryanbugden/marginSelectionTool#2]
Clicking on points and a SB + dragging isn’t fully solved yet. If you then click and drag mouse from the SB after making the selection, the points of deselected. If you click and drag from points, it works, but holding down Shift (to keep things from moving vertically) will kill the action.

**Comment by ryanbugden** [comment]
Need to figure out a way to _not_ deselect points when you `mouseDown` on SBs. (To behave more similarly to an actual point).

Maybe on `mouseUp`, point selection is stored. One `mouseDown`, if selecting SB, reassure the stored point selection remains selected (to counteract native deselection).

**Comment by typoman** [comment]
Why not add SBs to selection only with shiftDown? This is how points are added to selection too.

**Comment by ryanbugden** [comment]
Yeah that currently works still. `shiftDown` on _mouseDragged_ however was cancelling the whole operation of moving the SB. When I removed it, I was able to hit Shift once dragging to maintain only horizontal movement of points. Hope that makes sense.

**Comment by ryanbugden** [comment]
Oh I thought you were referring to 30c8717! Forget that.

What I meant was that I think this scenario should work:
Select SB → select points → click and drag SB = all pts and SB move together.

This works if you click and drag selected pts but not SB. See end of video [here](https://www.dropbox.com/s/6lnoq2kjibec10p/Screen%20Recording%202020-04-02%20at%2020.12.35.mov?dl=0).

**Comment by typoman** [comment]
Ow I got it. This is a tricky one, yeah maybe keeping the selection. I will see if I can do it with less effort!
